pandas/tests/frame/test_convert_to.py

```diff
@@ -71,6 +71,12 @@ def test_to_dict_timestamp(self):
         tm.assert_dict_equal(test_data_mixed.to_dict(orient='split'),
                              expected_split_mixed)
 
+    def test_to_dict_index_not_unique_with_index_orient(self):
```

Comment:
is this raise only happening on orient='index' or other orients as well?

Reply:
Yes

Reply:
Can you parametrize this function to test all of the orients? FWIW this probably should be allowed with `orient="table"` so curious how that behaves

Reply:
Hmm nevermind - I suppose this change is only applicable to `index` orients given that's the only one with the index value as a key of the dict